middle_end/flambda/un_anf.ml

```diff
@@ -122,43 +122,66 @@ let make_var_info (clam : Clambda.ulambda) : var_info =
        to traverse some subexpression. *)
-    | Uvar var -> add_use t var depth; environment_vars
+    | Uvar var -> add_use t var depth;
+    environment_vars
     | Uconst const ->
       (* The only variables that might occur in [const] are those in constant
          closures---and those are all bound by such closures.  It follows that
          [const] cannot contain any variables that are bound in the current
          scope, so we do not need to count them here.  (The function bodies
          of the closures will be traversed when this function is called from
          [Flambda_to_clambda.to_clambda_closed_set_of_closures].) *)
-      ignore_uconstant const; environment_vars
+      ignore_uconstant const;
+      environment_vars
     | Udirect_apply (label, args, dbg) ->
       ignore_function_label label;
-      List.fold_left (loop ~depth ~environment_vars) args;
-      ignore_debuginfo dbg; environment_vars
+      let environment_vars =
+        List.fold_left
+          (fun environment_vars arg -> loop ~depth ~environment_vars arg)
+          environment_vars
+          args
+      in
+      ignore_debuginfo dbg;
+      environment_vars
     | Ugeneric_apply (func, args, dbg) ->
       loop ~depth ~environment_vars func;
-      List.fold_left (loop ~depth ~environment_vars) args;
-      ignore_debuginfo dbg; environment_vars
+      let environment_vars =
+        List.fold_left
+          (fun environment_vars arg -> loop ~depth ~environment_vars arg)
+          environment_vars
+          args
+      in
+      ignore_debuginfo dbg;
+      environment_vars
     | Uclosure (functions, captured_variables) ->
-      List.fold_left (loop ~depth ~environment_vars) captured_variables;
-      List.iter (fun (
+      let environment_vars =
+        List.fold_left
+          (fun environment_vars captured_variables -> loop ~depth ~environment_vars captured_variables)
+          environment_vars
+          captured_variables
+      in
+      let environment_vars =
+      List.fold_left (fun (
         { Clambda. label; arity; params; return; body; dbg; env; } as clos) ->
-        let environment_vars =
-          (match closure_environment_var clos with
-           | None -> ()
-           | Some env_var ->
-            V.Set.add (VP.var env_var) environment_vars ) in
+          let environment_vars =
+            (match closure_environment_var clos with
+            | None -> environment_vars
+            | Some env_var ->
+              V.Set.add (VP.var env_var) environment_vars )
+          in
           ignore_function_label label;
           ignore_int arity;
           ignore_params_with_value_kind params;
           ignore_value_kind return;
           loop ~depth:(depth + 1) ~environment_vars:environment_vars body;
           ignore_debuginfo dbg;
-          ignore_var_option env)
-        functions
+          ignore_var_option env;
+          environment_vars)
+        in
+        functions;
```

Comment:
There's a bit of confusion here: the last `List.fold_left`, above, is missing some of its arguments, and `functions;` doesn't make much sense on its own as it's not a computations.